src/Components/SmartComponents/HybridInventory/ImmutableDevicesTab/ImmutableDevices.js

```diff
@@ -35,6 +35,7 @@ const ImmutableDevices = ({ intl, cveName, filterRuleValues, inventoryRef, heade
     ], 'inventory');
 
     const dispatch = useDispatch();
+    const navigate = useNavigate();
```

Comment:
You can use `const navigate = useInsightsNavigate('inventory')` here and the `navigate` only be `navigate(`/${systemId}?appName=vulnerabilities`);`